hibernate-core/src/main/java/org/hibernate/collection/spi/PersistentCollection.java

```diff
@@ -82,7 +84,8 @@ public interface PersistentCollection<E> extends LazyInitializable {
 	 * @param role The collection role
 	 * @param snapshot The snapshot state
 	 */
-	void setSnapshot(Object key, String role, Serializable snapshot);
+	// $ToBeRemoved$ @Nullables added on account of the setSnapshot calls in CollectionEntry
```

Comment:
This actually seems ok to me. All these parameters can be null, so IMO you can remove the comment.